### ISSUE TITLE
[BUG] This package works with TailwindCSS V4 and Nuxt V4, but keep reports "Cannot resolve default tailwindcss config path".

### ISSUE THREAD

**Issue by Hanning-Liu** [francoismassart/eslint-plugin-tailwindcss#418]
**Describe the bug**
 This package works with TailwindCSS V4 and Nuxt V4, but keep reports "Cannot resolve default tailwindcss config path".
**To Reproduce**
1. Go to '[the codesandbox workspace](https://codesandbox.io/p/live/a22ddb54-436b-41a5-a989-c6cd7cbd88ed)'
2. Open a 'terminal'
3. Input "pnpm lint"
4. Scroll down to '....'
5. See error/warning msg

**Expected behavior**
It doesn't look for the default tailwindcss config path.
**Screenshots**

<img width="1310" height="985" alt="Image" src="https://github.com/user-attachments/assets/ae21b986-6e1c-469d-9fd1-924f1ef772b8" />

**Environment (please complete the following information):**
 - OS: [Debian GNU/Linux 12]
 - Softwares + version used:
<img width="426" height="498" alt="Image" src="https://github.com/user-attachments/assets/6a1f7509-afe1-47a3-81a8-1fb21c6f7b45" />

**Additional context**

**eslint config file or live demo**
By providing a link to a live demo, a demo video or a github repo fixing the issue will be easier.

https://codesandbox.io/p/live/a22ddb54-436b-41a5-a989-c6cd7cbd88ed

**Comment by Hanning-Liu** [comment]
PLUS: This package will also disable my vscode OnSave actions:
```json
"editor.codeActionsOnSave": {
    "source.fixAll.eslint": "explicit",
    "source.organizeImports": "never"
  },
```

**Comment by DamianGlowala** [comment]
@Hanning-Liu your reproduction either doesn't exist or might be private

@francoismassart this affects me as well (it can't find `tailwind.config.ts`). Would you have any idea why it cannot be found?

**Comment by Hanning-Liu** [comment]
@DamianGlowala 
Updated the sandbox link.
https://codesandbox.io/p/live/a22ddb54-436b-41a5-a989-c6cd7cbd88ed

**Comment by Namielusi** [comment]
I had the same problem and I was able to solve it.

- Note that Tailwind V4 uses CSS files for configuration instead of the old `tailwind.config.js` file. The `eslint-plugin-tailwindcss` documentation doesn't reflect that.
- `4.0.0-beta.0` version of `eslint-plugin-tailwindcss` can't auto-detect path to the css file since it uses old tailwind API meant for `tailwind.config.js` that no longer exists in V4 https://github.com/francoismassart/eslint-plugin-tailwindcss/blob/29bee22a691d7280a2210b046b37802806a307f1/lib/util/settings.js#L4-L9

In order to fix the issue, you need to explicitly specify path to your root css file in your eslint config, for example:

```
import { dirname } from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);`

export default [{
  settings: {
    tailwindcss: {
      config: `${__dirname}/src/app/globals.css`,
    },
  },
}]
```

In my case, it didn't work with a relative path.

---

In order to keep using `tailwind.config.js` file, you need to use `@config` directive in your CSS file, for example:

```
@config "../tailwind.config.js";
```

**Comment by DamianGlowala** [comment]
@Namielusi thank you, this works for me!

**Comment by Hanning-Liu** [comment]
It works for me too! 😀

**Comment by Hanning-Liu** [comment]
But the VSCode OnSave actions are still not working after fixing the default tailwindcss config issue.
e.g. remove a semicolumn, then save, the eslint won't automatically bring the semicolumn back.

> PLUS: This package will also disable my vscode OnSave actions:
> 
> "editor.codeActionsOnSave": {
>     "source.fixAll.eslint": "explicit",
>     "source.organizeImports": "never"
>   },

**Comment by Hanning-Liu** [comment]
The VSCode OnSave actions issue fixed by running the cmds below:
```shell
pnpm add -D eslint-plugin-import
pnpm dedupe
```

I tried to reproduce the successful project in codesandbox but it fails since the ESLint extension cannot be changed to run in node.js 22.

CodeSandBox:
```
[Info  - 11:45:45 PM] ESLint server is starting.
[Info  - 11:45:45 PM] ESLint server running in node v16.20.2
[Info  - 11:45:45 PM] ESLint server is running.
[Info  - 11:45:45 PM] ESLint library loaded from: /project/workspace/node_modules/.pnpm/eslint@9.32.0/node_modules/eslint/lib/api.js
[Error - 11:45:45 PM] An unexpected error occurred:
[Error - 11:45:45 PM] SyntaxError: Unexpected token 'with'
    at ESMLoader.moduleStrategy (node:internal/modules/esm/translators:119:18)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:468:14)
[Error - 11:45:45 PM] An unexpected error occurred:
[Error - 11:45:45 PM] SyntaxError: Unexpected token 'with'
    at ESMLoader.moduleStrategy (node:internal/modules/esm/translators:119:18)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:468:14)

```

Local project on my laptop:
```
[Info  - 7:46:48 AM] ESLint server is starting.
[Info  - 7:46:48 AM] ESLint server running in node v22.15.1
[Info  - 7:46:49 AM] ESLint server is running.
[Info  - 7:46:49 AM] ESLint library loaded from: C:\Users\user\Desktop\New_folder\v4_autofix\node_modules\.pnpm\eslint@9.32.0_jiti@2.5.1\node_modules\eslint\lib\api.js
[@antfu/eslint-config] Detected running in editor, some rules are disabled.

```

I am going to close this issue.

**Comment by francoismassart** [comment]
Just got back from holidays, glad to read you found a solution.

I think you cannot point directly to a JS config using v4, you need to point to a css file which can contain a call to the config directive

https://tailwindcss.com/docs/functions-and-directives#config-directive

```css
@config "../../tailwind.config.js";
```

As I'm focusing on eslint-plugin-tailwindcss v4, I won't update the docs of v3 as it will get replaced by the v4.
